src/rest/RESTManager.js

```diff
@@ -7,34 +7,121 @@ const { Error } = require('../errors');
 const Collection = require('../util/Collection');
 const { Endpoints } = require('../util/Constants');
 
+/**
+ * Manages REST requests to the Discord API
+ */
 class RESTManager {
-  constructor(client, tokenPrefix = 'Bot') {
+  /**
+   *
+   * @param {Client} client The Discord client
+   * @param {?string} tokenPrefix The token prefix for the Authorization header
+   * @param {?string} token The token to perform requests with
+   */
+  constructor(client, tokenPrefix = 'Bot', token = client.token) {
+    /**
+     * The client who's configuration & utilities to use
+     * @type {Client}
+     * @readonly
+     * @private
+     */
     this.client = client;
+
+    /**
+     * The token used to perform requests
+     * @type {string}
+     * @readonly
+     * @private
+     */
+    this.token = token;
+
+    /**
+     * All current Requesthandlers
+     * @type {Collection<string, RequestHandler>}
+     * @readonly
+     * @private
+     */
     this.handlers = new Collection();
+
+    /**
+     * The token prefix for the Authorization header
+     * @type {string}
+     * @private
+     * @readonly
+     */
     this.tokenPrefix = tokenPrefix;
+
+    /**
+     * Whether or not to include the API version in requests
+     * @type {boolean}
+     * @private
+     * @readonly
+     */
     this.versioned = true;
+
+    /**
+     * The current global timeout, if any.
+     * @type {Promise<void>|null}
+     * @private
+     */
     this.globalTimeout = null;
+
     if (client.options.restSweepInterval > 0) {
       client.setInterval(() => {
         this.handlers.sweep(handler => handler._inactive);
       }, client.options.restSweepInterval * 1000);
     }
   }
 
+  /**
+   * Returns the route builder for crafting and eventually performing API requests.
+   * @example
+   * const manager = new RESTManager(client, 'Bearer', 'the bearer token');
+   * manager.api.users('@me').get().then(console.log);
+   * @example
+   * // returns an array of the authorized user's guilds
+   * manager.api.users('@me').guilds.get().then(console.log);
+   * @type {Object}
+   */
   get api() {
     return routeBuilder(this);
   }
 
+  /**
+   * Creates the Authorizaation header value
+   * @returns {string}
+   */
   getAuth() {
-    const token = this.client.token || this.client.accessToken;
-    if (token) return `${this.tokenPrefix} ${token}`;
+    if (this.token) return `${this.tokenPrefix} ${this.token}`;
     throw new Error('TOKEN_MISSING');
   }
 
+  /**
+   * Returns various functions for creating CDN links
+   * @type {Object}
```

Comment:
```suggestion
   * @type {object}
```

Reply:
Per https://jsdoc.app/tags-type.html, `object` is reserved for the type of typedefs